cloud/cluster/cluster.go

```diff
@@ -0,0 +1,64 @@
+package cluster
+
+import (
+	"fmt"
+	"io"
+)
+
+type NodeId string
+
+type Node interface {
+	Id() NodeId
```

Comment:
I think that's probably right, but we should wait until we're using it for real and can design it as we use it.

We might also have this be one of those cases where you expand on the interface, like, you might have some cluster implementations end up having two ports to talk to different services (e.g., http vs. thrift), so I don't think we can proscribe it too much just yet...